lib/scanner/blockqueue.go

```diff
@@ -94,40 +94,35 @@ func (ph *parallelHasher) hashFiles(ctx context.Context) {
 	defer ph.wg.Done()
 
 	for {
-		select {
```

Comment:
Why remove the outer select? I guess the intent is that the inbox will be closed, but it looks odd that we only check the context on one "side" of the processing stage?

Reply:
For one I somehow managed to commit the related changes in walk.go to another branch, making this change nonsensical anyway. My previous logic was to only check context on `outbox`/`finishedChan`, which is returned to the caller. That's unnecessary though, the only place we mustn't check context is the counter goroutine to not break the link between filesystem walking and hashing. Then wait on `ph.inbox` to be closed before closing `ph.outbox` to be sure filesystem walking finished when the returned channel gets closed.